lib/model/list_model.dart

```diff
@@ -21,47 +21,43 @@ class ListModel {
 
   Iterable<Item> itemsView() => items;
 
+  @ignore
+  Iterable<Item> get scheduledItems =>
+      itemsView().where((item) => item.isScheduled);
+
   // a zero-arg constructor is required for classes that are isar collections
   ListModel();
   ListModel.fromTitle(this.title) : labels = [];
 
   void init() {
-    reload();
-    // This ensures that labels is mutable
+    items.loadSync();
     labels = labels.toList();
+    updateTimersForScheduledItems();
   }
 
-  void reload() => items.loadSync();
-
-  Future<Iterable<Item>> searchItems(String searchQuery) {
-    final words = _parseSearchStr(searchQuery);
-    return items
-        .filter()
-        .allOf(words, (q, word) => q.valueContains(word, caseSensitive: false))
-        .findAll();
+  void updateTimersForScheduledItems({void Function(Item)? timerCallback}) {
```

Comment:
I don't really like this system. It seems like you override the item's timer callback using it?